### PR TITLE
add mariadb probe

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Implement #913
2. Add MariaDB probe

## How did you implement / how did you fix it  
1. Add mariadb library.
2. Update validation / json-schema 
3. Add guide probe documentation

## How to test  
1. run `npm run test` should all pass
2. create something like the  following `test.yaml`:
```yaml
notifications:
  - id: unique-id
    type: desktop

probes:
  - id: 'id-mariadb'
    name: Mariadb probe
    mariadb:
      - host: 172.17.0.1
        port: 3306
        username: user
        password: password
        database: mydb
    incidentThreshold: 2
    recoveryThreshold: 2
    alerts:
      - assertion: response.status != 200
        message: 'Alert! DB offline'

  - id: 'id mysql'
    name: Mysql Probe
    mysql:
      - host: 172.17.0.1
        port: 3306
        username: user
        password: password
        database: mydb
    incidentThreshold: 2
    recoveryThreshold: 2
    alerts:
      - assertion: response.status != 200
        message: 'Alert! DB offline'
```
3. run `bin/run-dev -c test.yaml -s -1` to probe the database. Should run with connection refused error
4. spin up a mariadb docker instance 
5. monika should detect the new mariadb connection
6. spin up a mysql docker instance instead, probing should still work

## Screenshot
Test result above in maridb mode
image

Test result in mysql mode
image

Notification created (desktop notification on linux)
image

Schema validation and suggestion
image

### CODE REVIEW

File: docs/src/pages/guides/probes.md [hunk 1]
```diff
@@ -75,6 +75,23 @@ Details of the field are given in the table below.
...
+### Mariadb Request
+
+To monitor connectivity and health of your MariaDB database, monika provides a MariaDB database probe, set it up like below:
+
+```yaml
+probes:
+  - id: 'id-mariadb'
+    name: maria_probe
+    description: testing maria db
+    maria:
```
Comment (kevin-hyperjump) [inline docs/src/pages/guides/probes.md:87]: I think it's better to use `mariadb` rather than `maria` to avoid ambiguity.

**Comment by dennypradipta** [comment]
Quick Q: Does this PR **explicitly** only supports MariaDB, or it could be used for MySQL too?

File: docs/src/pages/guides/probes.md [hunk 2]
```diff
@@ -75,6 +75,23 @@ Details of the field are given in the table below.
 | ping (optional)              | (boolean), If set true then send a PING to the specified url instead.                                                                                                                                                                                                                                                                                     |
 | allowUnauthorized (optional) | (boolean), If set to true, will make https agent to not check for ssl certificate validity                                                                                                                                                                                                                                                                |
 
+### Mariadb Request
+
+To monitor connectivity and health of your MariaDB database, monika provides a MariaDB database probe, set it up like below:
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:80]: To monitor the connectivity and the health of your MariaDB/MySQL database, you can use the following monika configuration

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/probe/index.ts [hunk 3]
```diff
@@ -319,6 +319,36 @@ export async function doProbe({
       }
     }
 
+    if (probe?.mysql) {
```
Comment (dennypradipta) [inline src/components/probe/index.ts:322]: Should this be `probe?.mysql || probe?.mariadb?`

Comment (sapiderman) [reply src/components/probe/index.ts]: good call.. done

**Review by dennypradipta — APPROVED** [review]

File: docs/src/pages/guides/probes.md [hunk 4]
```diff
@@ -75,6 +75,38 @@ Details of the field are given in the table below.
 | ping (optional)              | (boolean), If set true then send a PING to the specified url instead.                                                                                                                                                                                                                                                                                     |
 | allowUnauthorized (optional) | (boolean), If set to true, will make https agent to not check for ssl certificate validity                                                                                                                                                                                                                                                                |
 
+### Mariadb Request
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:78]: MariaDB or MySQL Probe

**Review by nicnocquee — CHANGES_REQUESTED** [review]

**Pull request merged**